addons/website_slides_survey/controllers/slides.py

```diff
@@ -3,52 +3,32 @@
 
 from odoo.addons.website_slides.controllers.main import WebsiteSlides
 
+from odoo import http
 from odoo.http import request
 
+import werkzeug.utils
+import werkzeug.exceptions
+
 
 class WebsiteSlides(WebsiteSlides):
 
+    @http.route(['/slides_survey/slide/get_certification_url'], type='http', auth='user', website=True)
+    def slide_get_certification_url(self, slide_id, **kw):
+        fetch_res = super(WebsiteSlides, self)._fetch_slide(slide_id)
+        if fetch_res.get('error'):
+            raise werkzeug.exceptions.NotFound()
+        slide = fetch_res['slide']
+        if slide.channel_id.is_member:
+            slide.action_set_viewed()
```

Comment:
We should remove the creation of the user_input from 'action_viewed', and do it in `_generate_certification_url`